### PR TITLE
DAOS-2721 control: Reorganize binaries under cmd/

### PR DESCRIPTION
**Pull request by mjmac**
When a Go project produces multiple binaries, it is a common
convention to place each binary's main.go in a subdirectory
of cmd/ named after the produced binary. e.g. cmd/daos_server

This organization convention allows for easy identification of
which binaries could be produced from the project, and takes
advantage of a property of the Go compiler which automatically
produces a binary named after the directory where main.go is
located.

### CODE REVIEW

File: src/control/server/server.go [hunk 1]
```diff
@@ -35,12 +35,6 @@ import (
 	"google.golang.org/grpc"
 )
 
-func main() {
```
Comment (jbrosenz) [inline src/control/server/server.go]: Is there no longer a need to perform the os.Exit(1) on error?  Or did it move somewhere I may have missed?

Comment (tanabarr) [reply src/control/server/server.go]: https://github.com/daos-stack/daos/pull/664/files#diff-bb8c5675ada3c31026ea839e31fdbf64R13

Comment (mjmac) [reply src/control/server/server.go]: Yeah, the idea here is to make an actual server package and move the main function into cmd/daos_server/main.go.

In the short term, this is mostly just cosmetic, but it lays groundwork for making some of this code reusable across multiple binaries and other reorganization possibilities.

**Comment by mjmac** [comment]
> looks good overall, and something I've been meaning to get to. seems to be duplication with control/cmd/dmg/_.go & control/cmd/dmg/dmg/_.go and the same with agent and drpc_test.

Whoops, good catch. I missed that in my local review before pushing. Fixed, thanks.

**Comment by mjmac** [comment]
> 
> 
> Are the git blame histories going to be lost by the renames and other file moves? If so, I think we want to talk about it to see if that matters enough to anyone.

git --follow still works.

**Pull request merged**